tools/build_env/build_env.sh

```diff
@@ -40,18 +43,18 @@ if ! which docker > /dev/null; then
 fi
 
 if [[ $BUILD = true ]]; then
-  readonly DOCKER_ROOT="$(dirname $0)"
-  docker build --rm -t $IMAGE_NAME $DOCKER_ROOT
+  readonly DOCKER_ROOT="$(dirname $0)/$TAG"
```

Comment:
I couldn't find anything in Docker's documentation regarding tags and versioning; on the other hand, there's a bunch of writing concerning tags and `latest`:
 * https://medium.com/@mccode/the-misunderstood-docker-tag-latest-af3babfd6375#.mwo5rkru4
 * http://container-solutions.com/docker-latest-confusion/

In short, `latest` is a poor name because it behaves like **`default`**.

That said, we can easily switch to images having names of the form `uproxy/build-whatever` - a little less elegant in my mind but PTAL.